### PR TITLE
Skip some tests in aarch64 CI

### PR DESCRIPTION
**Pull request by emcastillo**
Some of the routines here rely in scipy functions that don't return the same results in x86/arm. Seems that the CUDA routines are written to mach the results in x86 causing several tests to fail.

https://github.com/scipy/scipy/issues/20158
https://github.com/scipy/scipy/issues/20160

### CODE REVIEW

File: tests/cupyx_tests/scipy_tests/special_tests/test_polygamma.py [hunk 1]
```diff
@@ -42,6 +44,9 @@ def test_scalar(self, xp, scp, dtype):
         return scp.special.polygamma(
             dtype(2.), dtype(1.5)).astype(numpy.float32)
 
+    @pytest.mark.xfail(
+        platform.processor() == "aarch64",
+        reason="aarch64 scipy does not match cupy/x86 see Scipy")
```
Comment (jakirkham) [inline tests/cupyx_tests/scipy_tests/special_tests/test_polygamma.py:49]: Is this meant to have an issue #?

File: cupyx/fallback_mode/fallback.py [hunk 2]
```diff
@@ -321,7 +321,12 @@ def _update_numpy_array(self):
                     if self._numpy_array is None:
                         self._numpy_array = base._numpy_array.view(type=_type)
                         self._numpy_array.shape = self._cupy_array.shape
-                        self._numpy_array.strides = self._cupy_array.strides
+                        # In some scenarios, the numpy array strides may be
```
Comment (kmaehashi) [inline cupyx/fallback_mode/fallback.py:324]: Can I have an example to reproduce this?

**Comment by leofang** [comment]
Note: This PR (#8411) and #8410 were patches needed when we worked on #8442 and ran the full test suite on G+H. None of these patches are specific to G+H, I believe they were just oversights due to lack of aarch64 CIs.

**Comment by leofang** [comment]
@emcastillo could you take a look at the questions above when you get a chance? 🙂 Thanks!

File: tests/cupyx_tests/scipy_tests/special_tests/test_polygamma.py (hunk 1 above)
Comment (emcastillo) [reply tests/cupyx_tests/scipy_tests/special_tests/test_polygamma.py:49]: Sorry, I think it was https://github.com/scipy/scipy/issues/20159
Will update it

File: cupyx/fallback_mode/fallback.py (hunk 2 above)
Comment (emcastillo) [reply cupyx/fallback_mode/fallback.py:324]: Actually this error seems to be gone now, probably it was some kind of numpy/cupy incompatibility when I tested it the first time.

Comment (emcastillo) [reply cupyx/fallback_mode/fallback.py:324]: Sorry, this is related to #8442, When running in a system with unified memory we get the following error

```FAILED tests/cupyx_tests/fallback_mode_tests/test_fallback.py::TestInplaceSpecialMethods::test_put_along_axis - ValueError: strides is not compatible with available memory
FAILED tests/cupyx_tests/fallback_mode_tests/test_fallback.py::TestArrayVariants::test_inplace - ValueError: strides is not compatible with available memory


 def _update_numpy_array(self):
        """
        Updates _numpy_array from _cupy_array.
        To be executed before calling numpy function.
    

        if self._supports_cupy:
            # cupy-compatible
            if base is None:
             _array)
            else:
                if not base._remember_numpy:

                        self._numpy_array.shape = self._cupy_array.shape
>                       self._numpy_array.strides = self._cupy_array.strides
E                       ValueError: strides is not compatible with available memory

cupyx/fallback_mode/fallback.py:324: ValueError
```

I think we can remove this fix from here and add it in the corresponding PR, or another option is to skip fallback mode tests when using the system allocator.

I discussed it a while ago with @leofang but is fallback mode being currently used? with the extremely high coverage that CuPy has, maybe it is worth deprecating it

**Review by kmaehashi — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks all! 🙏

Looks like CI is green 😄

**Pull request merged**